File: backend/components/metiers/threed/fixtures/core/validators/fixtures_validator.py
"""
fixtures_validator.py - Validation avancée des fixtures Threed (Python)
"""
def is_valid_3d_model(model):
    return bool(model and isinstance(model.get('id'), str) and isinstance(model.get('vertices'), list) and isinstance(model.get('faces'), int))

def is_valid_user(user):
    return bool(user and isinstance(user.get('id'), str) and isinstance(user.get('role'), str))

File: backend/components/metiers/voyage/fixtures/core/validators/fixtures_validator.js
// fixtures_validator.js - Validation avancée des fixtures Voyage

module.exports = {
  isValidvoyageModel: (model) => {
    return model && typeof model.id === 'string' && Array.isArray(model.vertices) && Array.isArray(model.faces);
  },
  isValidUser: (user) => {
    return user && typeof user.id === 'string' && typeof user.role === 'string';
  }
};
